File: Backend/src/main/java/com/example/bus_reservation_system/entity/Details.java
package com.example.bus_reservation_system.entity;

import jakarta.persistence.Column;
import lombok.Data;

@Data
public class Details {
    private long id;
    private long reservation_id;
    private String name;
    private String gender;
    private int age;
    private String stateOfResidence;
    private String contactDetails;
    private int seatNumber;
}

File: Frontend/src/app/Models/details.ts
export class Details{
    id: number;
    reservation_id: number;
    name: string;
    age: number;
    gender: string;
    stateOfResidence: string;
    contactDetails: string;
    seatNumber:number;
    selected?: boolean;
    
    constructor(
      name: string='',
      age: number=0,
      gender: string='',
      state_of_residence: string='',
      contact_details: string='',
      seatNumber:number=0,
      id: number,
      reservation_id: number,
      selected?: boolean
    ) {
      this.name = name;
      this.age = age;
      this.gender = gender;
      this.stateOfResidence = state_of_residence;
      this.contactDetails = contact_details;
      this.id = id; // Optional
      this.reservation_id = reservation_id;
      this.seatNumber=seatNumber; 
      this.selected=selected;
    }
}